cli/cmd/common/utils.go

```diff
@@ -1,20 +1,81 @@
 package common
 
 import (
+	"embed"
 	"fmt"
 	"os"
 	"strings"
 
 	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart/loader"
 	helmCLI "helm.sh/helm/v3/pkg/cli"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 )
 
 const (
 	DefaultReleaseName      = "consul"
 	DefaultReleaseNamespace = "consul"
+	chartFileName           = "Chart.yaml"
+	valuesFileName          = "values.yaml"
+	templatesDirName        = "templates"
+	TopLevelChartDirName    = "consul"
 )
 
+// ReadChartFiles reads the chart files from the embedded file system, and loads their contents into
+// []*loader.BufferedFile. This is a format that the Helm Go SDK functions can read from to create a chart to install
+// from. The names of these files are important, as there are case statements in the Helm Go SDK looking for files named
+// "Chart.yaml" or "templates/<templatename>.yaml", which is why even though the embedded file system has them named
+// "consul/Chart.yaml" we have to strip the "consul" prefix out.
+func ReadChartFiles(chart embed.FS, chartDirName string) ([]*loader.BufferedFile, error) {
+	var chartFiles []*loader.BufferedFile
+
+	bytes, err := chart.ReadFile(fmt.Sprintf("%s/%s", chartDirName, chartFileName))
```

Comment:
nit: we can use `filepath.Join` for these kinds of expressions (`filepath.Join(chartDirName, chartFileName)`)